problem_builder/public/js/plot.js

```diff
@@ -0,0 +1,218 @@
+function PlotBlock(runtime, element) {
+
+    // Plot
+
+    // Define margins
+    var margins = {top: 20, right: 20, bottom: 20, left: 20};
+
+    // Define width and height of SVG viewport
+    var width = 440;
+    var height = 440;
+
+    // Define dimensions of plot area
+    var plotWidth = width - margins.left - margins.right;
+    var plotHeight = height - margins.top - margins.bottom;
+
+    // Preselect target DOM element for plot.
+    // This is necessary because when using a CSS selector,
+    // d3.select will select the *first* element that matches the selector (in document traversal order),
+    // which leads to unintended consequences when multiple plot blocks are present.
+    var plotTarget = $(element).find('.sb-plot').get(0);
+
+    // Create SVG viewport with nested group for plot area
+    var svgContainer = d3.select(plotTarget)
+        .append("svg")
+        .attr("width", width)
+        .attr("height", height)
+        .append("g")
+        .attr("transform", "translate(" + margins.left + ", " + margins.right + ")");
+
+    // Create scales to use for axes and data
+    var xScale = d3.scale.linear()
+        .domain([0, 100])
+        .range([0, plotWidth]);
+
+    var yScale = d3.scale.linear()
+        .domain([100, 0])
+        .range([0, plotHeight]);
+
+    // Create axes
+    var xAxis = d3.svg.axis()
+        .scale(xScale)
+        .orient("bottom");
+
+    var yAxis = d3.svg.axis()
+        .scale(yScale)
+        .orient("left");
+
+    // Create SVG group elements for axes and call the xAxis and yAxis functions
+    var xAxisGroup = svgContainer.append("g")
+        .attr("transform", "translate(0, " + plotHeight / 2 + ")")
+        .call(xAxis);
+
+    var yAxisGroup = svgContainer.append("g")
+        .attr("transform", "translate(" + plotWidth / 2 + ", 0)")
+        .call(yAxis);
+
+    // Buttons
+
+    var defaultButton = $('.plot-default', element);
```

Comment:
Nit: With this many var statements in sequence, you may want to use the comma syntax:

```
var defaultButton = $('.plot-default', element),
    averageButton = $('.plot-average', element),
    var quadrantsButton = $('.plot-quadrants', element);
```

But that can mean diffs by adding a new line are slightly bigger. Up to you.